### PR TITLE
Fix broken links from issue #46336

### PR DESCRIPTION
**Pull request by brentmackinnon**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Fixes a handful of 404'd links from https://github.com/woocommerce/woocommerce/issues/46336 

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. check that each new link works and makes sense in context. 

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Fix a few broken links.

#### Comment 

</details>

### CODE REVIEW

File: docs/contributing/common-issues.md
Comment (Konamiman) [inline docs/contributing/common-issues.md]: Typo:

```suggestion
Also see [this document](https://woocommerce.com/document/woopayments/testing-and-troubleshooting/sandbox-mode/).
```

File: docs/payments/payment-gateway-api.md
Comment (lsinger) [inline docs/payments/payment-gateway-api.md]: This could be a good opportunity to remove the title that just repeats the URL from the link markdown -- or is there a special reason why we need that? 

```suggestion
It is also important that your gateway class extends the WooCommerce base gateway class, so you have access to important methods and the [settings API](https://developer.woocommerce.com/docs/settings-api/):
```

(Also see one other link in this PR.)

File: docs/payments/payment-gateway-api.md
Comment (lsinger) [inline docs/payments/payment-gateway-api.md]: See above. 

```suggestion
Use this method to set `$this->form_fields` - these are options you'll show in admin on your gateway settings page and make use of the [WC Settings API](https://developer.woocommerce.com/docs/settings-api/).
```

File: docs/quality-and-best-practices/writing-high-quality-testing-instructions.md
Comment (lsinger) [inline docs/quality-and-best-practices/writing-high-quality-testing-instructions.md]: The URL `https://woocommerce.com/document/woocommerce-payments/testing-and-troubleshooting/-mode/` does not work for me.

**Review by Konamiman — APPROVED** [review]
Approving, but waiting for @lsinger 's review before merging.

**Review by lsinger — APPROVED** [review]
To ensure all CI checks pass, I had to trigger the automation for adding changelog files, fix a markdown linter issue, regenerate the docs manifest, and then resolve a merge conflict. Good to go now. 👍

**Pull request merged**
